### PR TITLE
feat(openai): Implement OpenAI o1 thinking mode with budget configuration (#98)

### PR DESCRIPTION
**Pull request by JGoP-L**
## AgentScope-Java Version

1.0.2-SNAPSHOT

### Background & Purpose
This PR implements support for OpenAI's o1 and o1-mini models with advanced reasoning capabilities, specifically the **thinking mode** feature. The thinking mode allows models to perform extended reasoning before generating responses, significantly improving accuracy on complex tasks.

Issue: #98 - "Support OpenAI o1 thinking mode with budget configuration"

#### Core Implementation
1. **OpenAIChatModel** (`src/main/java/io/agentscope/core/model/OpenAIChatModel.java`)
   - Added `enableThinking` configuration parameter (Boolean, supports three states: true/false/null)
   - Added `toolsHelper` member variable for centralized parameter management
   - Implemented `applyThinkingIfAvailable()` method to apply thinking configuration
   - Implemented `isThinkingEnabled()` method to check if thinking mode is enabled
   - Modified `doStream()` to:
     - Check tool compatibility with thinking mode (mutual exclusion)
     - Apply thinking configuration before other options (to prevent override)
     - Disable streaming when thinking mode is enabled
   - Added `enableThinking()` method to Builder class

2. **OpenAIToolsHelper** (`src/main/java/io/agentscope/core/formatter/openai/OpenAIToolsHelper.java`)
   - Implemented `applyThinking()` method to:
     - Handle thinking budget configuration (with fallback to defaultOptions)
     - Build OpenAI-compliant thinking parameter: `{"type": "enabled", "budget_tokens": <number>}`
     - Apply configuration only when budget > 0
   - Enhanced parameter filtering in:
     - `applyAdditionalBodyParams()`: Filter "thinking" key to prevent user override
     - `applyAdditionalQueryParams()`: Filter null values
     - `applyAdditionalHeaders()`: Filter null values
     - `applyTools()`: Filter null parameter values

3. **OpenAIResponseParser** (`src/main/java/io/agentscope/core/formatter/openai/OpenAIResponseParser.java`)
   - Implemented `extractThinking()` method to:
     - Use reflection to access thinking field from ChatCompletionMessage
     - Handle multiple data types (String, Optional, wrapper objects)
     - Implement graceful degradation for SDK versions without thinking support
   - Implemented `isDefaultObjectToString()` helper for robust pattern matching
   - Complete exception handling for reflection operations

4. **Test Suite** (`src/test/java/io/agentscope/core/model/OpenAIThinkingModeTest.java`)
   - 15 comprehensive unit tests covering:
     - Model creation with thinking enabled/disabled/null
     - Thinking budget configuration
     - Thinking parameter application with options priority
     - Response parsing and thinking content extraction
     - Edge cases: empty thinking, null values, special characters

#### Key Features
- ✅ Support for o1 and o1-mini models
- ✅ Configurable thinking budget (tokens)
- ✅ Parameter priority handling: `options > defaultOptions`
- ✅ Mutual exclusion: Thinking mode disables streaming and tool calling
- ✅ Parameter protection: Users cannot override thinking configuration via additionalBodyParams
- ✅ Null value filtering to prevent SDK errors
- ✅ Reflection-based compatibility for SDK versions without public thinking API
- ✅ Complete exception handling and graceful degradation

#### Bug Fixes
- Fixed duplicate calls to `isThinkingEnabled()` - now reuses cached computed value
- Improved logging in `applyThinking()` to include cases where thinking mode is not applied

### Testing & Validation

**Test Results:**

**Test Coverage:**
- Model creation tests: 3 cases (with/without thinking)
- Configuration tests: 3 cases (budget handling, priority, streaming)
- Parameter application tests: 5 cases (thinking parameter application)
- Response extraction tests: 4 cases (content extraction and edge cases)

**Validation Checklist:**
- ✅ All existing tests pass (1420/1420)
- ✅ No new test failures or regressions
- ✅ Code formatted with `mvn spotless:apply`
- ✅ No compilation errors
- ✅ No checkstyle violations in modified files
- ✅ Javadoc comments complete and follow project conventions

#### 1. Build and Test
```bash
# Format code
mvn spotless:apply

# Run all tests (including new thinking mode tests)
mvn test -DexcludedGroups=e2e -am -pl agentscope-core

# Run only thinking mode tests
mvn test -Dtest="OpenAIThinkingModeTest" -am -pl agentscope-core

### CODE REVIEW

File: agentscope-core/src/main/java/io/agentscope/core/formatter/openai/OpenAIResponseParser.java [hunk 1]
```diff
@@ -308,4 +318,83 @@ protected ChatResponse parseChunkResponse(ChatCompletionChunk chunk, Instant sta
...
+     * so we use reflection to access it if available.
+     *
+     * @param message ChatCompletionMessage from OpenAI API
+     * @return Thinking content string, or null if not present
+     */
+    protected String extractThinking(ChatCompletionMessage message) {
+        try {
+            // First, try to use a public getter if it exists
+            try {
+                Field thinkingField = ChatCompletionMessage.class.getDeclaredField("thinking");
```
Comment (AlbumenJ) [inline agentscope-core/src/main/java/io/agentscope/core/formatter/openai/OpenAIResponseParser.java]: Does the OpenAI Java SDK currently support this field?

I checked the latest source code in `ChatCompletionMessage.kt` but couldn't locate it: [https://github.com/openai/openai-java/blob/main/openai-java-core/src/main/kotlin/com/openai/models/chat/completions/ChatCompletionMessage.kt](https://github.com/openai/openai-java/blob/main/openai-java-core/src/main/kotlin/com/openai/models/chat/completions/ChatCompletionMessage.kt)

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: agentscope-core/src/main/java/io/agentscope/core/formatter/openai/OpenAIResponseParser.java (hunk 1 above)
Comment (JGoP-L) [reply agentscope-core/src/main/java/io/agentscope/core/formatter/openai/OpenAIResponseParser.java]: > Does the OpenAI Java SDK currently support this field?
> 
> I checked the latest source code in `ChatCompletionMessage.kt` but couldn't locate it: https://github.com/openai/openai-java/blob/main/openai-java-core/src/main/kotlin/com/openai/models/chat/completions/ChatCompletionMessage.kt

是的 我又去看了sdk。
Chat Completions API 不支持 O 系列推理模型。O1/O3 等推理模型必须使用 Responses API（而非 Chat Completions API），通过 ResponseCreateParams.builder().reasoning(Reasoning.builder().effort(...).budget(...).build()) 传入推理配置。
所以说我前面的修改不对，我重新修改。

Comment (AlbumenJ) [reply agentscope-core/src/main/java/io/agentscope/core/formatter/openai/OpenAIResponseParser.java]: 可以看一下目前其他模型通过 OpenAI 转发以后怎么带的 Thinking block，比如 openrouter、dashscope、deepseek 等

Comment (JGoP-L) [reply agentscope-core/src/main/java/io/agentscope/core/formatter/openai/OpenAIResponseParser.java]: > 可以看一下目前其他模型通过 OpenAI 转发以后怎么带的 Thinking block，比如 openrouter、dashscope、deepseek 等

如果使用Chat Completions API 是不太可行的，我看官方推荐的也是Responses API。我如果使用OpenAI 用 Responses API → 返回 ThinkingBlock  你觉得这样可以吗？

Comment (AlbumenJ) [reply agentscope-core/src/main/java/io/agentscope/core/formatter/openai/OpenAIResponseParser.java]: Responses API 这个得单独开一个实现了，默认的话还是最好走 chat，这个是兼容性最好的模式

Comment (JGoP-L) [reply agentscope-core/src/main/java/io/agentscope/core/formatter/openai/OpenAIResponseParser.java]: > Responses API 这个得单独开一个实现了，默认的话还是最好走 chat，这个是兼容性最好的模式

OpenAIChatModel（默认）- Chat API 零改动，完全兼容，用户无感知
OpenAIReasoningModel（可选）- Responses API 完全独立，仅供需要推理功能的用户使用
用两个独立的 Model class，
这样才能最大化兼容性（默认用户零改动）和可用性（需要推理的用户能用上） 我是这么认为的。
